tpm2/constants.go

```diff
@@ -0,0 +1,181 @@
+	AlgSHA512    Algorithm = 0x000D
+	AlgNULL      Algorithm = 0x0010
+	AlgRSASSA    Algorithm = 0x0014
+	AlgRSAES     Algorithm = 0x0015
+	AlgRSAPSS    Algorithm = 0x0016
+	AlgOAEP      Algorithm = 0x0017
+	AlgECDSA     Algorithm = 0x0018
+	AlgECDH      Algorithm = 0x0019
+	AlgECDAA     Algorithm = 0x001A
+	AlgECC       Algorithm = 0x0023
+	AlgCTR       Algorithm = 0x0040
+	AlgOFB       Algorithm = 0x0041
+	AlgCBC       Algorithm = 0x0042
+	AlgCFB       Algorithm = 0x0043
+	AlgECB       Algorithm = 0x0044
+	AlgLAST      Algorithm = 0x0044
+	AlgKEYEDHASH Algorithm = 0x0008
+)
+
+// SessionType defines the type of session created in StartAuthSession.
+type SessionType uint8
+
+// Supported session types.
+const (
+	SessionHMAC   SessionType = 0x00
+	SessionPolicy SessionType = 0x01
+	SessionTrial  SessionType = 0x03
+)
+
+// KeyProp is a bitmask used in Attributes field of key templates. Individual
+// flags should be OR-ed to form a full mask.
+type KeyProp uint32
+
+// Key properties.
+const (
+	FlagFixedTPM            KeyProp = 0x00000002
+	FlagFixedParent         KeyProp = 0x00000010
+	FlagSensitiveDataOrigin KeyProp = 0x00000020
+	FlagUserWithAuth        KeyProp = 0x00000040
+	FlagAdminWithPolicy     KeyProp = 0x00000080
+	FlagRestricted          KeyProp = 0x00010000
+	FlagDecrypt             KeyProp = 0x00020000
+	FlagSign                KeyProp = 0x00040000
+
+	FlagSealDefault   = FlagFixedTPM | FlagFixedParent
+	FlagSignerDefault = FlagSign | FlagRestricted | FlagFixedTPM |
+		FlagFixedParent | FlagSensitiveDataOrigin | FlagUserWithAuth
+	FlagStorageDefault = FlagDecrypt | FlagRestricted | FlagFixedTPM |
+		FlagFixedParent | FlagSensitiveDataOrigin | FlagUserWithAuth
+)
+
+// A Handle is an identifier for TPM objects.
+type Handle uint32
+
+// Reserved Handles.
+const (
+	HandleOwner           Handle = 0x40000001
+	HandleRevoke          Handle = 0x40000002
+	HandleTransport       Handle = 0x40000003
+	HandleOperator        Handle = 0x40000004
+	HandleAdmin           Handle = 0x40000005
+	HandleEK              Handle = 0x40000006
+	HandleNull            Handle = 0x40000007
+	HandleUnassigned      Handle = 0x40000008
+	HandleLockout         Handle = 0x4000000A
+	HandleEndorsement     Handle = 0x4000000B
+	HandlePlatform        Handle = 0x4000000C
+	PasswordSessionHandle Handle = 0x40000009
+)
+
+// Capability identifies some TPM property or state type.
+type Capability uint32
+
+// TPM Capabilies.
+const (
+	CapabilityAlgs          Capability = 0x00000000
+	CapabilityHandles       Capability = 0x00000001
+	CapabilityCommands      Capability = 0x00000002
+	CapabilityPPCommands    Capability = 0x00000003
+	CapabilityAuditCommands Capability = 0x00000004
+	CapabilityPCRs          Capability = 0x00000005
+	CapabilityTPMProperties Capability = 0x00000006
+	CapabilityPCRProperties Capability = 0x00000007
+	CapabilityECCCurves     Capability = 0x00000008
+	CapabilityAuthPolicies  Capability = 0x00000009
+)
+
+type structureTag uint16
```

Comment:
Why is `structureTag` not exported and the others are? Even though it's not supported, I'd like to have a comment about the type so that readers can have some context. That comment could also explain why it's not exported.